### PR TITLE
Add demo_android Dockerfile

### PR DESCRIPTION
**Pull request by grwlf**
This PR aims at adding demo_android Dockerfile to the set of other environments. At this stage, I was able to walk through jvm manual and build `app/android_deploy` with this image so I consider it is ready for review. The discussion branch is available at https://discuss.tvm.ai/t/dockerfile-for-mobile-development/629/4

### CODE REVIEW

**Comment by tqchen** [comment]
@eqy can you review this?

**Comment by eqy** [comment]
Do we want some simple utility scripts included that can help with apps that use OpenCL and Vulkan (e.g., download required libraries/headers, and pull `libOpenCL.so`, `libvulkan.so`, etc from common paths on phones? )

**Comment by grwlf** [comment]
Note that I originally planned to support environments for building `apps/android_deploy` and `apps/android_rpc` by this PR. Of cause, it doesn't mean we should stop there. I don't have much experience in mobile development, and I feel It would be extremely helpful to have a demo application involving Vulcan, together with formally-specified environment to build it.

**Comment by grwlf** [comment]
At this stage I am able to build both `apps/android_deploy` and `apps/android_rpc`, so I consider this PR is ready.  @eqy could you please look into it? I have some doubts regarding naming since 'Dockerfile.demo_android` can't be actually used to run android demo, only to build it.

File: docker/install/ubuntu_install_androidsdk.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+extras;m2repository;com;android;support;constraint;constraint-layout;1.0.2
+extras;m2repository;com;android;support;constraint;constraint-layout-solver;1.0.2
+lldb;2.3
+platforms;android-26
+platforms;android-27
+tools
+ndk-bundle
+EOF
+
+touch /root/.android/repositories.cfg
```
Comment (eqy) [inline docker/install/ubuntu_install_androidsdk.sh]: This seems to error out when I try to build the image; does it required a `.android` to already exist?

Comment (grwlf) [reply docker/install/ubuntu_install_androidsdk.sh]: Thanks. It does, indeed. Fixed.

**Comment by tqchen** [comment]
@eqy can you update your review or explicitly approve it?

**Comment by eqy** [comment]
@grwlf Looks like it builds for me now; can you provide a little documentation on how to properly use the docker file e.g., to build an app?

**Comment by grwlf** [comment]
I originally followed instructions from [android_deploy/README.md](https://github.com/dmlc/tvm/tree/master/apps/android_deploy), the `Build APK` section. I do realize that the procedure is still complex, below I will try to collect the bare minimum: 

1. Build TVM4J
```
# In TVM project root folder:
$ make jvmpkg
$ make jvminstall
```

2. Prepare config
```
$ cd apps/android_deploy/app/src/main/jni
$ cp make/config.mk .
$ edit config.mk   # If needed. I kept default values.
```

3. Build APKs
```
$ echo $ANDROID_HOME # Should already be set from Docker's /etc/profile
$ cd apps/android_deploy
$ gradle clean build
$ dev_tools/gen_keystore.sh
$ dev_tools/sign_apk.sh
```

5. tvmdemo-release.apk should be ready for uploading to device

**Comment by grwlf** [comment]
@merrymercy  On our phone, it takes approx 4 sec. to classify the image. The last attempt took 3947ms. Probably, we should try enabling OpenCL first and then go for autotuning?

**Comment by merrymercy** [comment]
With latest master branch, I think we can make the performance significantly better if we compile the saved model using `with nnvm.compiler.build_config(opt_level=3):` and `target = "llvm -device=arm_cpu"`

But for this PR, completing the doc on instructions is enough to merge.

**Comment by grwlf** [comment]
I've updated the README.md of `apps/android_deploy` which seems to be the main source of information about the topic. Could you please review it?

**Comment by tqchen** [comment]
@eqy @merrymercy please take another look and https://docs.tvm.ai/contribute/code_review.html#approve-and-request-changes-explicitly

**Comment by tqchen** [comment]
Thanks @grwlf @merrymercy @eqy , this is now merged

**Comment by tqchen** [comment]
The docker binary has been build and upstreamed to tvmai/demo-android

**Pull request merged**